libreoffice-core/sw/source/uibase/uno/colorizer.cxx

```diff
@@ -699,14 +715,19 @@ void resetColors(CancelFlag& cancelFlag, rtl::Reference<SwXTextDocument> doc)
             pInternalCursor = comphelper::getFromUnoTunnel<SwXTextCursor>(xBodyCursor);
             if (!pInternalCursor)
                 continue;
-            pDoc->getIDocumentContentOperations().InsertPoolItem(*pInternalCursor->GetPaM(),
-                                                                 sResetColor);
+            UpdateTextColorDirectly(pDoc, pInternalCursor->GetPaM(), sResetColor);
         }
     }
 }
 
+
+
+
 void colorize(CancelFlag& cancelFlag, rtl::Reference<SwXTextDocument> doc)
 {
+
+    SolarMutexGuard aGuard;
```

Comment:
Rather than fussing with this, I think that maybe we should just comment this out and make a TODO.
It's very likely that only one thread will ever use this anyway, after I'm done with the ELOK stuff.

```suggestion
    // TODO: Make thread safe? SolarMutexGuard locks out the cancel flag, so this isn't simple
```

Reply:
Please verify that cancel works after this.